File: samples/SampleCounterV3/Presenters/CounterPresenter.cs
using SampleCounterV2.Models;
using SampleCounterV2.Views;

namespace SampleCounterV2.Presenters;

public sealed class CounterPresenter(ICounterView view, Counter initialState)
{
    private Counter _state = initialState;

    public void StartApplication(string containerName)
    {
        var renderer = new Renderer(containerName);
        renderer.Render(view);

        view.OnButtonClick = OnButtonClicked;
        view.ShowCounter(_state.Count);
    }

    private void OnButtonClicked()
    {
        _state = _state.Increment();
        view.ShowCounter(_state.Count);
    }
}

File: targets/js/sample-counter-v3/lib/presenters/counter-presenter.js
import { Renderer } from "#/views";
export class CounterPresenter {
    _view;
    _state;
    constructor(view, initialState) {
        this._view = view;
        this._state = initialState;
    }
    startApplication(containerName) {
        const renderer = new Renderer(containerName);
        renderer.render(this._view);
        this._view.onButtonClick = this.onButtonClicked.bind(this);
        this._view.showCounter(this._state.count);
    }
    onButtonClicked() {
        this._state = this._state.increment();
        this._view.showCounter(this._state.count);
    }
}
